codec.go

```diff
@@ -56,20 +59,43 @@ func setupCodec() error {
 				max = addr
 			}
 		}
+		addrs = append(addrs, int(addr))
```

Comment:
I'm worried that addrs can grow if the offset value is very large.

Reply:
Yes, you are right, it will grow following the length of `offset`, I can add a limit for this to ensure it doesn't grow beyond maxAcceptableTypeAddrRange.